pootle/apps/pootle_project/models.py

```diff
@@ -420,6 +423,13 @@ def delete(self, *args, **kwargs):
 
         directory.delete()
 
+    def clean(self):
+        if not self.code:
+            raise ValidationError('A code must be provided.')
```

Comment:
[As already commented](https://github.com/translate/pootle/issues/5114#issuecomment-241642828), `blank=False` is the default and applies to forms.